test/manifests/app.yaml

```diff
@@ -30,9 +31,23 @@ spec:
       value: database/creds/$(ROLE)
     - name: VAULT_KUBERNETES_ROLE
       value: database-access
+    - name: VAULT_KUBERNETES_TOKEN_PATH
+      value: /var/run/secrets/tokens/test-application
     - name: DB_NAME
       value: postgres
     - name: DB_HOST
       value: postgres
     - name: TLS_MODE
-      value: disable
\ No newline at end of file
+      value: disable
+    volumeMounts:
+    - mountPath: /var/run/secrets/tokens
+      name: vault-token
+  volumes:
+  - name: vault-token
+    projected:
+      sources:
+      - serviceAccountToken:
+          path: test-application
+          expirationSeconds: 7200
+          audience: api
```

Comment:
Let's have a separate audience `vault` for Vault. This way only the projected tokens can access Vault (must be configured in Vault). In this case, it doesn't matter (w.r.t. access to Vault) whether the default token is mounted in the main container.